src/Autofac/Builder/BuildCallbackService.cs

```diff
@@ -0,0 +1,77 @@
+﻿// This software is part of the Autofac IoC container
+// Copyright © 2019 Autofac Contributors
+// https://autofac.org
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Autofac.Builder
+{
+    /// <summary>
+    /// Represent a collection of callbacks that can be run after a container or scope is 'built'.
+    /// </summary>
+    internal class BuildCallbackService
+    {
+        private List<Action<ILifetimeScope>> Callbacks { get; set; } = new List<Action<ILifetimeScope>>();
```

Comment:
Should this be a member variable rather than a property? Unclear if the compiler just optimizes that out anyway.

Reply:
The compiler will optimise that to a directly field access, but it should just be a member; I had an older variant that allowed external access to the callbacks, but the property hung around after the change.